packages/macro/src/index.ts

```diff
@@ -110,54 +121,132 @@ function macro({ references, state, babel, config }: MacroParams) {
     }
   })
 
+  if (needsUseLinguiImport) {
+    addImport(babel, body, useLinguiImportModule, useLinguiImportName)
+  }
+
   if (needsI18nImport) {
-    addImport(babel, state, i18nImportModule, i18nImportName)
+    addImport(babel, body, i18nImportModule, i18nImportName)
   }
 
   if (jsxNodes.size) {
-    addImport(babel, state, TransImportModule, TransImportName)
+    addImport(babel, body, TransImportModule, TransImportName)
   }
 }
 
 function reportUnsupportedSyntax(path: NodePath, e: Error) {
   throw path.buildCodeFrameError(
-    `Unsupported macro usage. Please check the examples at https://lingui.dev/ref/macro#examples-of-js-macros. 
+    `Unsupported macro usage. Please check the examples at https://lingui.dev/ref/macro#examples-of-js-macros.
  If you think this is a bug, fill in an issue at https://github.com/lingui/js-lingui/issues
- 
+
  Error: ${e.message}`
   )
 }
 
+/**
+ * Pre-process useLingui macro
+ * 1. Get references to destructed t function for macro processing
+ * 2. Transform usage to non-macro useLingui
+ *
+ * @returns Array of paths to useLingui's t macro
+ */
+function processUseLingui(
+  path: NodePath,
+  newIdentifier: string
+): NodePath[] | null {
+  if (!path.parentPath.parentPath.isVariableDeclarator()) {
+    reportUnsupportedSyntax(
+      path,
+      new Error(
+        `\`useLingui\` macro must be used in variable declaration.
+        
+ Example:
+
+ const { t } = useLingui()
+`
+      )
+    )
+    return null
+  }
+
+  const varDec = path.parentPath.parentPath.node
+  const _property = t.isObjectPattern(varDec.id)
+    ? varDec.id.properties.find(
+        (
+          property
+        ): property is t.ObjectProperty & {
+          key: t.Identifier
+          value: t.Identifier
+        } =>
+          t.isObjectProperty(property) &&
+          t.isIdentifier(property.key) &&
+          t.isIdentifier(property.value) &&
+          property.key.name == "t"
+      )
+    : null
+
+  if (!_property) {
+    reportUnsupportedSyntax(
+      path.parentPath.parentPath,
+      new Error(
+        `Must destruct \`t\` when using \`useLingui\` macro, i.e:
+ const { t } = useLingui()
+ or
+ const { t: _ } = useLingui()
+ `
+      )
+    )
+    return null
+  }
+
+  if (t.isIdentifier(path.node)) {
```

Comment:
here, I'd like to ask again for more help to anyone reading this. What does this condition cover? Can we give a name to `t.isIdentifier(path.node)` so that the condition is less "magical"?

Reply:
Mostly of the `isXXXX` from babel types act as type guard, giving us a way to get access to properties of a concrete type. In this case we need to access `name` of `path.node`, to which you wouldn't have had access to without this check.

Not sure how we can "name" it.

Reply:
well, a condition always represents some scenario. If the scenario is given a name or a comment I think more people will understand it.

// rename to standard useLingui

and what are we renaming?

Feel free to ignore though